profile/src/main/java/com/scottlogic/deg/profile/dto/ProfileSchemaFileLoader.java

```diff
@@ -39,22 +39,34 @@ public ProfileSchemaFileLoader(ProfileSchemaValidator validator) {
 
     @Override
     public void validateProfile(File profileFile, URL schemaUrl) {
+        String profile;
         String schema;
+
         if (schemaUrl == null) {
             throw new ValidationException("Null Schema");
         }
-        try {
-            byte[] data = Files.readAllBytes(profileFile.toPath());
-            String profile = readAllLines(data).stream().collect(Collectors.joining(System.lineSeparator()));
-
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(schemaUrl.openStream()))) {
-                schema = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
 
-            validator.validateProfile(profile, schema);
+        try {
+            profile = readProfile(profileFile);
+            schema = readSchema(schemaUrl);
         } catch (IOException e) {
             throw new ValidationException(e.getClass() + " when looking for schema with URL " + schemaUrl);
         }
+
+        validator.validateProfile(profile, schema);
```

Comment:
optional. -
may as well put this line into the try, and get rid of the separate declarations at the top of the method